swift-mode-repl.el

```diff
@@ -751,17 +779,103 @@ PROCESS-IDENTIFIER is the process ID."
     (goto-char comint-last-input-end)
     (search-forward expected-output nil t)))
 
+;;;###autoload
+(defun swift-mode:debug-ios-device (project-directory
+                                    device-identifier
+                                    scheme
+                                    codesigning-folder-path)
+  "Run debugger on an iOS app in the PROJECT-DIRECTORY.
+Run it for the iOS local device DEVICE-IDENTIFIER for the given SCHEME.
+DEVICE-IDENTIFIER is the device identifier of the iOS simulator.
+SCHEME is the name of the project scheme in Xcode.
+CODESIGNING-FOLDER-PATH is the path of the codesigning folder in Xcode
+build settings."
+  (swift-mode:build-ios-app project-directory device-identifier scheme)
+  (swift-mode:run-repl
+   (append
+    (swift-mode:command-string-to-list swift-mode:ios-deploy-executable)
+    (list "--debug" "--no-wifi" "--bundle" codesigning-folder-path))
+   nil t))
+
+;;;###autoload
+(defun swift-mode:debug-ios-simulator (project-directory
+                                       device-identifier
+                                       scheme
+                                       codesigning-folder-path
+                                       product-bundle-identifier)
+  "Run debugger on an iOS app in the PROJECT-DIRECTORY.
+Run it for the iOS simulator DEVICE-IDENTIFIER for the given SCHEME.
+DEVICE-IDENTIFIER is the device identifier of the iOS simulator.
+SCHEME is the name of the project scheme in Xcode.
+CODESIGNING-FOLDER-PATH is the path of the codesigning folder used in Xcode
+build settings.
+PRODUCT-BUNDLE-IDENTIFIER is the path of the product bundle identifier used
```

Comment:
`product-bundle-identifier` is a bundle identifier like `org.example.some_app`. Not a path.